memtable/write_buffer_manager.cc

```diff
@@ -88,9 +90,16 @@ void WriteBufferManager::ReserveMemWithCache(size_t mem) {
   while (new_mem_used > cache_rep_->cache_allocated_size_) {
     // Expand size by at least 256KB.
     // Add a dummy record to the cache
-    Cache::Handle* handle;
+    Cache::Handle* handle = nullptr;
     cache_rep_->cache_->Insert(cache_rep_->GetNextCacheKey(), nullptr,
                                kSizeDummyEntry, nullptr, &handle);
+    // We keep the handle even if insertion fails and a null handle is
+    // returned, so that when memory shrinks, we don't release extra
+    // entries from cache.
+    // Ideallly we should prevent this allocation from happening if
+    // this insertion fails. However, the callers to this code path
+    // are not able to handle failures properly. We'll need to improve
+    // it in the future.
```

Comment:
@siying can you clarify what happens when the insert fails? Does the write just stall or will it need to be retried?

Reply:
Or does it just ignore the strict capacity limit for writes, but still enforce it for "real" cache blocks

Reply:
If this insert fails, we just don't cost the memory to block cache, which means, total memory will exceed the configured cap.